utils/code_generator/main.cpp

```diff
@@ -0,0 +1,173 @@
+#include <iostream>
+#include <string>
+#include <sstream>
+#include "tiramisu_code_generator.h"
+#include <random>
+#include <cstdlib>
+
+using namespace std;
+
+
+void read_inputs(int *nb_codes, int *nb_stages, string *default_type_tiramisu, string *default_type_wrapper, double *assignment_prob, double *assignment_input_prob, double *conv_prob, double *same_padding_prob, vector<int> *computations_dimensions, 
+                int *nb_inputs,  vector <int> *var_nums, int *offset);
+int main() {
+    int nb_codes, nb_stages, nb_inputs, offset;
+    vector <int> computations_dimensions, var_nums;
+    double assignment_prob, assignment_input_prob, conv_prob, same_padding_prob;
+    string defaut_type_tiramisu, default_type_wrapper;
+    read_inputs(&nb_codes, &nb_stages, &defaut_type_tiramisu, &default_type_wrapper, &assignment_prob, &assignment_input_prob, &conv_prob, &same_padding_prob, &computations_dimensions, &nb_inputs, &var_nums, &offset);
+
+
+    double num, *padding_probs = new double[1], *computations_probs = new double[2];  
+
+    padding_probs[0] = same_padding_prob;
+
+    computations_probs[0] = assignment_prob;
+    computations_probs[1] = assignment_input_prob;
+
+    for (int i = 0; i < nb_codes; ++i){
+        num = (double) rand() / (RAND_MAX);
```

Comment:
```suggestion
        num = (double) rand() / (RAND_MAX);
        if (num < conv_prob){
```

This functionality is very common.  Instead of doing like this, can you create a function that allows easily to future users to generate random number and test it.  Something like

if (generate_random_value() < conv_prob)
    ...
else
   ....